tycho-core/src/main/java/org/eclipse/tycho/core/maven/TychoMavenLifecycleParticipant.java

```diff
@@ -234,11 +234,13 @@ private void validateUniqueBaseDirs(List<MavenProject> projects) throws MavenExe
         }
     }
 
+    private static final Set<String> CLEAN_PHASES = Set.of("pre-clean", "clean", "post-clean");
+
     private boolean disableLifecycleParticipation(MavenSession session) {
         // command line property to disable Tycho lifecycle participant
         return "maven".equals(session.getUserProperties().get("tycho.mode"))
                 || session.getUserProperties().containsKey("m2e.version")
-                || session.getGoals().equals(List.of("clean")); // disable for clean-only session
+                || (session.getGoals().size() == 1 && CLEAN_PHASES.contains(session.getGoals().get(0))); // disable for clean-only session
```

Comment:
```suggestion
                || CLEAN_PHASES.containsAll(session.getGoals());
```

Reply:
That makes it more general and simpler. Thanks.